sdk/keyvault/azure-keyvault-administration/tests/test_backup_client.py

```diff
@@ -82,24 +112,66 @@ def test_selective_key_restore(self, container_uri, sas_token):
         delete_poller.wait()
         key_client.purge_deleted_key(key_name)
 
+    @ResourceGroupPreparer(random_name_enabled=True, use_cache=True)
+    @StorageAccountPreparer(random_name_enabled=True)
+    @BlobContainerPreparer()
+    def test_backup_client_polling(self, container_uri, sas_token):
+        if not self.is_live:
+            pytest.skip("Poller requests are incompatible with vcrpy in playback")
 
-def test_continuation_token():
-    """Methods returning pollers should accept continuation tokens"""
+        # backup the vault
+        backup_client = KeyVaultBackupClient(self.managed_hsm["url"], self.credential)
+        backup_poller = backup_client.begin_backup(container_uri, sas_token)
+
+        # create a new poller from a continuation token
+        token = backup_poller.continuation_token()
+        rehydrated = backup_client.begin_backup(container_uri, sas_token, continuation_token=token)
 
-    expected_token = "token"
-    mock_generated_client = mock.Mock()
+        # check that pollers and polling methods behave as expected
+        assert backup_poller.status() == "InProgress"
+        assert not backup_poller.done() or backup_poller.polling_method().finished()
+        assert rehydrated.status() == "InProgress"
+        assert not rehydrated.done() or rehydrated.polling_method().finished()
 
-    backup_client = KeyVaultBackupClient("vault-url", object())
-    backup_client._client = mock_generated_client
-    backup_client.begin_restore("storage uri", "sas", continuation_token=expected_token)
-    backup_client.begin_backup("storage uri", "sas", continuation_token=expected_token)
-    backup_client.begin_restore("storage uri", "sas", key_name="key", continuation_token=expected_token)
+        backup_poller.polling_method().update_status()
+        assert backup_poller.status() == "InProgress"
+        rehydrated.polling_method().update_status()
+        assert rehydrated.status() == "InProgress"
+
+        backup_operation = backup_poller.result()
+        assert backup_poller.status() == "Succeeded" and backup_poller.polling_method().status() == "Succeeded"
+        rehydrated_operation = rehydrated.result()
+        assert rehydrated.status() == "Succeeded" and rehydrated.polling_method().status() == "Succeeded"
+        assert backup_operation.folder_url == rehydrated_operation.folder_url
 
-    for method in ("begin_full_backup", "begin_full_restore_operation", "begin_selective_key_restore_operation"):
-        mock_method = getattr(mock_generated_client, method)
-        assert mock_method.call_count == 1
-        _, kwargs = mock_method.call_args
-        assert kwargs["continuation_token"] == expected_token
+        # rehydrate a poller with a continuation token of a completed operation
+        late_rehydrated = backup_client.begin_backup(container_uri, sas_token, continuation_token=token)
+        assert late_rehydrated.status() == "InProgress"
```

Comment:
Is this still true? I would expect the poller to be done because `begin_backup` created it with the latest status.

Reply:
This is surprisingly still true, though I don't understand why. The pipeline response we get when we check the status inside `KeyVaultBackupClient.begin_backup` shows "Succeeded" in the response, but the poller (and polling method) we get back have an "InProgress" status until `update_status` gets called. My guess would be that newly-created pollers start off with this status by default until they do any actual polling. I probably shouldn't test for this though, since it seems like more of a base implementation concern

Reply:
`OperationResourcePolling.set_initial_status` assumes the status of a 20x response is "InProgress". Naively, seems like it ought to call its own `get_status(response)` instead. But I can imagine cases in which an initial response doesn't carry a status. So, this is something we would fix in `KeyVaultBackupClientPolling.set_initial_status`